hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java

```diff
@@ -2567,6 +2573,17 @@ public ReportProcedureDoneResponse reportProcedureDone(RpcController controller,
     return ReportProcedureDoneResponse.getDefaultInstance();
   }
 
+  private void throwOnOldMasterStartCode(Stream<Long> masterStartCodeStream)
+    throws MasterNotRunningException {
+    long masterStartCodeFromProc = masterStartCodeStream.max(Long::compare).orElse(-1L);
+    // -1 is less than any possible MasterStartCode
+
+    if (masterStartCodeFromProc > server.getStartcode()) {
```

Comment:
So if we throw exception here what would be the befit of it. Will client refresh the stub and go to currently active master?

Reply:
yes. [Here](https://github.com/apache/hbase/blob/2a4203f6fcb63b4563ea660fb03b0b9648b944ac/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java#L2356) is the logic